### ISSUE TITLE
can the doi be toggled? 

### ISSUE THREAD

**Issue by zimmerst** [pkgw/yahapj#4]
Hi -
first off, this is great. Indeed just what I was looking for. So my bib file has numerous doi entries but when running (pdf)latex with bibtex and the mentioned hyperref option, the doi is not put in my bibliography. So is there a place where I need to enable the doi field? Thanks!

**Comment by pkgw** [comment]
It depends on what you're aiming for. This style does not explicitly show the DOIs at the ends of references, if that's what you want; it only uses them to make hyperlink destinations. That choice is not going to change since that follows the ApJ style, but it wouldn't be too much work to make a hacked version that adds the DOI written out as well.

Or are you trying to accomplish something different?

**Comment by zimmerst** [comment]
AH -- okay - in that case I misunderstood the intention. So for the DOI i'd need to hack the style to add another field to the usual output, correct? I've briefly looked at the yahapj.bst file but couldn't quite figure out how to add this info.

**Comment by pkgw** [comment]
Yes, the BST language is pretty unusual. It might work well to try to modify the maybe_doi_link_end function to look like this (**untested code!!!**):

```
FUNCTION {maybe.doi.link.end}
{ % Expects some citation text to be on the stack
 doi empty$ url empty$ and
  { }
  { "} [DOI:" doi * "]" * * }
 if$
}
```

The best documentation of the language of which I'm aware is [Tame the BeaST](http://tug.ctan.org/info/bibtex/tamethebeast/ttb_en.pdf).
